src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -2517,7 +2363,10 @@ public function rollbackTransaction()
 
         try {
             $this->inTransaction = false;
-            $this->namespaces = array();
+
+            // reset namespaces
+            $this->namespaces->exchangeArray(array());
+            $this->getNamespacesObject();
```

Comment:
the problem with the reference is that using the ArrayObject does not trigger a reload. but i guess we need to go with this for now and hope to find a solution to pass the NamespaceRegistry in 2.0